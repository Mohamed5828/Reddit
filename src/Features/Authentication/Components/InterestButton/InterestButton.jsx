import React from "react";
import { BaseButton } from "./InterestButton.styled";

/**
 * InterestButton component that is used in AfterSignUp modals
 * @param {string} children The text of the button
 * @param {Array} otherProps Props to know the shape of the button
 * @returns {React.Component}  InterestButton component that is used in AfterSignUp modals
 */
const InterestButton = ({ children, ...otherProps }) => {
  return <BaseButton {...otherProps}>{children}</BaseButton>;
};

export default InterestButton;
