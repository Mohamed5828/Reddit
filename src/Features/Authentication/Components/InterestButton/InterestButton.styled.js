import styled from "styled-components";

const fontFamily = "'IBM Plex Sans', sans-serif;";

export const BaseButton = styled.button`
  line-height: 40px;
  background-color: ${(p) =>
    p.selected
      ? "#9c8282"
      : ({ theme }) => theme.color.mainInputBackgroundColor};

  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: Noto Sans, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  border-radius: ${(p) => (p.page ? "5px" : "999px")};
  color: ${(p) => (p.selected ? "white" : "black")};
  height: 40px;
  padding: 0 16px;
  font-family: ${fontFamily};
  text-transform: unset;
  margin: 5px 3px;
  width: fit-content;
  cursor: pointer;
  align-items: center;

  svg {
    margin-right: 5px;
  }
`;
