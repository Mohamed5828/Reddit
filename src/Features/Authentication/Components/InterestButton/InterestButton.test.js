import { shallow } from "enzyme";

import InterestButton from "./InterestButton"; 

describe("InterestButton Component", () => {
  it("this is a test for InterestButton Component", () => {
    expect(shallow(<InterestButton />)).toMatchSnapshot();
  });
});
