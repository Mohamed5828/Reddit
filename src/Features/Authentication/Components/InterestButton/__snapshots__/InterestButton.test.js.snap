// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestButton Component this is a test for InterestButton Component 1`] = `<styled.button />`;
